sat-extension/src/main/java/org/openhab/tools/analysis/report/SummaryReportHtmlGenerator.java

```diff
@@ -0,0 +1,155 @@
+        File latestMergeResultCopy = new File(summaryReportDirectory, SUMMARY_XML_FILE_NAME);
+
+        FileChannel mergeLockFileChannel = null;
+        FileChannel summaryLockFileChannel = null;
+
+        try {
+            // Acquire the merge and summary locks
+            mergeLockFileChannel = acquireFileLock(summaryReportDirectory, MERGE_LOCK_FILE_NAME);
+            if (!latestMergeResult.exists()) {
+                return null;
+            }
+            summaryLockFileChannel = acquireFileLock(summaryReportDirectory, SUMMARY_LOCK_FILE_NAME);
+
+            // Copy merge.xml to summary.xml which is used for generating the report
+            Files.copy(latestMergeResult, latestMergeResultCopy);
+
+            // Release the merge lock so plugin reporting goals executed in parallel can keep merging
+            closeFileChannel(mergeLockFileChannel);
+
+            File latestSummaryReport = new File(summaryReportDirectory, SUMMARY_REPORT_FILE_NAME);
+            run(CREATE_HTML_XSLT, latestMergeResultCopy, latestSummaryReport);
+
+            if (!latestMergeResultCopy.delete()) {
+                logger.error("Unable to delete file: " + latestMergeResultCopy.getAbsolutePath());
+            }
+
+            return latestSummaryReport;
+        } catch (InterruptedException | FileLockInterruptionException e) {
+            Thread.currentThread().interrupt();
+        } catch (IOException e) {
+            throw new IllegalStateException("Exception while acquiring lock file", e);
+        } finally {
+            closeFileChannel(mergeLockFileChannel);
+            closeFileChannel(summaryLockFileChannel);
+        }
+        return null;
+    }
+
+    private void closeFileChannel(FileChannel fileChannel) {
+        if (fileChannel != null) {
+            try {
+                fileChannel.close();
+            } catch (IOException e) {
+                logger.error("Exception while closing file channel: " + fileChannel, e);
+            }
+        }
+    }
+
+    private FileChannel acquireFileLock(final String summaryReportDirectory, final String fileName)
+            throws InterruptedException, FileLockInterruptionException {
+        File mergeLockFile = new File(summaryReportDirectory, fileName);
+        try {
+            return ReportUtil.acquireFileLock(mergeLockFile);
+        } catch (IOException e) {
+            throw new IllegalStateException("Exception while acquiring lock file: " + mergeLockFile, e);
+        }
+    }
+
+    private void run(final String xslt, final File input, final File output) {
+        try (FileOutputStream outputStream = new FileOutputStream(output)) {
+            if (logger.isDebugEnabled()) {
+                logger.debug(MessageFormat.format("{0}  > {1} >  {2}", input, xslt, output));
+            }
+
+            // Process the Source into a Transformer Object
+            final InputStream inputStream = contextClassLoader.getResourceAsStream(xslt);
```

Comment:
I will move them to a try-with-resources statement to be sure there won't be any (future) issues. Usually there is no explicit need to close the streams.

See: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html

> Streams have a BaseStream.close() method and implement AutoCloseable, but nearly all stream instances do not actually need to be closed after use. Generally, only streams whose source is an IO channel (such as those returned by Files.lines(Path, Charset)) will require closing. Most streams are backed by collections, arrays, or generating functions, which require no special resource management. (If a stream does require closing, it can be declared as a resource in a try-with-resources statement.)